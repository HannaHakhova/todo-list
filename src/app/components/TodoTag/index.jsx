import React from 'react';
import {Label, Icon} from 'semantic-ui-react';
import {string, shape, func, number} from 'prop-types';

const TodoTag = ({todoTag, handleDeleteTodoTag}) =>
    (
        <Label as='a' size='tiny' tag color='blue'> {todoTag.title}
            <Icon name='delete' onClick={() => handleDeleteTodoTag(todoTag.bindingId)} />
        </Label>
    );


TodoTag.propTypes = {
    todoTag: shape({
        id: string.isRequired,
        title: string.isRequired,
        bindingId: number.isRequired}).isRequired,
    handleDeleteTodoTag: func.isRequired
};

export default TodoTag;