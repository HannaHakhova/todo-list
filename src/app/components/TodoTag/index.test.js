/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import {shallow} from 'enzyme';
import TodoTag from './index';

describe('TodoTag', () => {
    it('matches expected snapshot)', () => {
        expect(shallow(<TodoTag
            handleDeleteTodoTag={() => {}}
            todoTag={{id: '123456', title: 'test', bindingId: 4}} />)).toMatchSnapshot();
    });

    it('calls handleDeleteTodoTag when it is clicked', () => {
        const handleDeleteTodoTag = jest.fn();
        const wrapper = shallow(<TodoTag
            handleDeleteTodoTag={handleDeleteTodoTag}
            tag={{id: '123456', title: 'test', bindingId: 4}} />);

        wrapper.find('Icon').simulate('click');

        expect(handleDeleteTodoTag).toBeCalledWith('123456');
    });
});