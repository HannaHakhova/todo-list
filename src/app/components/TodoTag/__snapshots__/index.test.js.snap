// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoTag matches expected snapshot) 1`] = `
<Label
  as="a"
  color="blue"
  size="tiny"
  tag={true}
>
   
  test
  <Icon
    as="i"
    name="delete"
    onClick={[Function]}
  />
</Label>
`;
